networkd/network_server.ml

```diff
@@ -1012,13 +1013,31 @@ module PVS_proxy = struct
     open S.PVS_proxy
 
     let path = ref "/opt/citrix/pvsproxy/socket/pvsproxy"
-
-	let do_call call =
-		try
-			Jsonrpc_client.with_rpc ~path:!path ~call ()
-		with e ->
-			error "Error when calling PVS proxy: %s" (Printexc.to_string e);
-			raise PVS_proxy_connection_error
+    
+    let do_call call =
+            let timebox ~timeout f =
+                let fd_in, fd_out = Unix.pipe () in
```

Comment:
If the `f` hangs forever, the `fd_out` will leak, so we should close `fd_out` once timeouted.
`Helpers.timebox` has the same issue. https://github.com/xapi-project/xen-api/blob/master/ocaml/xapi/helpers.ml#L1133